cmd/rokka/cli/copyall/copyall.go

```diff
@@ -0,0 +1,102 @@
+package copyall
+
+import (
+	"sync"
+
+	"github.com/rokka-io/rokka-go/rokka"
+)
+
+//Options general options for the whole copy-all command
+type Options struct {
+	SourceOrganization      string
+	DestinationOrganization string
+	DryRun                  bool
+	Concurrency             int
+	NoProgress              bool
+}
+
+// CopyResult contains the result of the operatio
+type CopyResult struct {
+	RokkaHash string
+	Error     error
+}
+
+// StartWorkers starts Copy Workers
+func StartWorkers(options Options, client *rokka.Client, images chan string, results chan CopyResult) {
+	waitGroup := sync.WaitGroup{}
+	waitGroup.Add(options.Concurrency)
+
+	// Start workers for image copy
+	for i := 0; i < options.Concurrency; i++ {
+		go func() {
+			defer waitGroup.Done()
+			copyWorker(client, images, results, options.SourceOrganization, options.DestinationOrganization, options.DryRun)
+		}()
+	}
+
+	// Start a go-routine to to close the result channel as soon as all workers are done
+	go func() {
+		waitGroup.Wait()
+		close(results)
+	}()
+
+}
+
+func copyWorker(
+	client *rokka.Client,
+	imageFiles chan string,
+	results chan CopyResult,
+	sourceOrg string,
+	destinationOrg string,
+	dryRun bool,
+) {
+	for hash := range imageFiles {
+		result := CopyResult{
+			RokkaHash: hash,
+		}
+
+		if !dryRun {
+			result.Error = executeRokkaCopy(client, hash, sourceOrg, destinationOrg)
+		}
+
+		results <- result
+
```

Comment:
in go you usually do not much empty lines, that's slightly different from other languages ;)

Reply:
removed some ;)